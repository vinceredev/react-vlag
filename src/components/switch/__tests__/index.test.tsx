import React from "react";
import renderer from "react-test-renderer";
import Switch from "../index";

it("renders correctly", () => {
  const tree = renderer.create(<Switch />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders checked true correctly", () => {
  const tree = renderer.create(<Switch checked={true} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders disabled true correctly", () => {
  const tree = renderer.create(<Switch disabled={true} />).toJSON();
  expect(tree).toMatchSnapshot();
});