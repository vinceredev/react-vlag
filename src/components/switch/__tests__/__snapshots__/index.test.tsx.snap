// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders checked true correctly 1`] = `
<div
  className="vlag-switch"
>
  <label
    className="vlag-switch__label"
  >
    <div
      className="vlag-switch__item vlag-switch__item-active "
    >
      <div
        className="vlag-switch__item__box"
      />
      <div
        className="vlag-switch__item__dots"
      />
    </div>
  </label>
</div>
`;

exports[`renders correctly 1`] = `
<div
  className="vlag-switch"
>
  <label
    className="vlag-switch__label"
  >
    <div
      className="vlag-switch__item  "
    >
      <div
        className="vlag-switch__item__box"
      />
      <div
        className="vlag-switch__item__dots"
      />
    </div>
  </label>
</div>
`;

exports[`renders disabled true correctly 1`] = `
<div
  className="vlag-switch"
>
  <label
    className="vlag-switch__label"
  >
    <div
      className="vlag-switch__item  vlag-switch__item-disabled"
    >
      <div
        className="vlag-switch__item__box"
      />
      <div
        className="vlag-switch__item__dots"
      />
    </div>
  </label>
</div>
`;
